drivers/serial/uart_native_posix.c

```diff
@@ -216,16 +348,48 @@ static int np_uart_0_init(const struct device *dev)
 	d = (struct native_uart_status *)dev->data;
 
 	if (IS_ENABLED(CONFIG_NATIVE_UART_0_ON_OWN_PTY)) {
-		int tty_fn = open_tty(d, DT_INST_LABEL(0),
-				      auto_attach);
+		int tty_fn = open_tty(DT_INST_LABEL(0), auto_attach);
 
-		d->in_fd = tty_fn;
-		d->out_fd = tty_fn;
-		np_uart_driver_api_0.poll_in = np_uart_tty_poll_in;
+		if (tty_fn < 0) {
+			EXIT("Could not open ptty for UART_0\n");
+		} else {
+			d->in_fd = tty_fn;
+			d->out_fd = tty_fn;
+		}
+	} else if (IS_ENABLED(CONFIG_NATIVE_UART_0_ON_DEVICE)) {
+		int tty_fn = open(uart0_pathname, O_RDWR | O_NOCTTY);
+
+		if (tty_fn < 0) {
+			if (!uart0_pathname) {
+				EXIT("--uart_0_device not set\n");
```

Comment:
Shouldn't this test be done before using it in `open()` above?

Reply:
you're probably right. uart0_pathname == NULL is a bit of a corner case of the various things that could be wrong with the pathname so I pulled it out as a separate error message. open(NULL) returns error on Linux and I expect most POSIX targets but apart from messing up the program flow a bit it wouldn't hurt to check in advance.